apps/visualization/views.py

```diff
@@ -7,45 +7,44 @@
 from django.db import connection
 from django_redis import get_redis_connection
 from rest_framework import response
-from utils import str_to_bool
+from utils import str_to_bool, check_page_limit
 from .models import (
     DataCountryLevelMostRecent,
     DataCountryLevelPublic,
     DataGranularPublic,
     DataCountryLevelPublicContext,
 )
-from .serializers import DataCountryLevelMostRecentSerializer
+from .serializers import DataCountryLevelMostRecentSerializer, SourceListAggSerializer
 from .rest_filters import (
     DataCountryLevelMostRecentFilter,
     BaseExportFilter,
+    SourceListAggFilter,
 )
 from rest_framework.pagination import LimitOffsetPagination
 from rest_framework.generics import ListAPIView
 from rest_framework.views import APIView
 
 
-class ContextIndicatorsViews(ListAPIView):
+class BasePaginatedApiView(ListAPIView):
+    default_limit = 10
+    pagination_class = LimitOffsetPagination
+
+
+class ContextIndicatorsViews(BasePaginatedApiView):
     '''
-    Returns most recent inidcator values for a country level.
+    Returns most recent indicator values for a country level.
     It accepts String query parameters such as emergency, region, iso3, type, thematic,
     topic as filter.
     It also accepts query parameters such as limit and offset to request number of objects
     in a page.
     Comma separated string query parameters can be given for topic to filter according
     to multiple topic.
     '''
-    default_limit = 10
     serializer_class = DataCountryLevelMostRecentSerializer
     filterset_class = DataCountryLevelMostRecentFilter
-    pagination_class = LimitOffsetPagination
 
     def get_queryset(self):
-        if self.request.query_params.get('limit'):
-            limit = int(self.request.query_params.get('limit'))
-            if limit > settings.OPEN_API_MAX_PAGE_LIMIT:
-                raise ValidationError(
-                    {'error': f'Limit must be less or equal to {settings.OPEN_API_MAX_PAGE_LIMIT}'}
-                )
+        check_page_limit(self)
```

Comment:
Looks like there is `max_limit` https://www.django-rest-framework.org/api-guide/pagination/#configuration_1 Maybe we can use that instead

Reply:
@thenav56 I tried it but it does not work.